src/components/SignUpForm/index.test.js

```diff
@@ -0,0 +1,409 @@
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test@test.co",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/^password/i), {
+      target: {
+        value: "password",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/repeat password/i), {
+      target: {
+        value: "password",
+      },
+    });
+
+    fireEvent.click(screen.getByRole("checkbox"));
+
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(1);
+    expect(mockSubmit).not.toBeCalled();
+    expect(screen.getByRole("textbox", { name: /firstName/i }).value).toBe(
+      "name"
+    );
+    expect(screen.getByRole("textbox", { name: /lastName/i }).value).toBe(
+      "surname"
+    );
+    expect(screen.getByRole("textbox", { name: /email/i }).value).toBe(
+      "test@test.co"
+    );
+    expect(screen.getByPlaceholderText(/^password/i).value).toBe("password");
+    expect(screen.getByPlaceholderText(/repeat password/i).value).toBe(
+      "password"
+    );
+    expect(screen.getByRole("checkbox").checked).toBe(true);
+  });
+
+  it("should display min length error when password is short", async () => {
+    const MIN_CHAR = 8;
+    fireEvent.input(screen.getByRole("textbox", { name: /firstName/i }), {
+      target: {
+        value: "name",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /lastName/i }), {
+      target: {
+        value: "surname",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test@mail.com",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/^password/i), {
+      target: {
+        value: `${"a".repeat(MIN_CHAR - 1)}`,
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/repeat password/i), {
+      target: {
+        value: `${"a".repeat(MIN_CHAR - 1)}`,
+      },
+    });
+
+    fireEvent.click(screen.getByRole("checkbox"));
+
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(1);
+    expect(mockSubmit).not.toBeCalled();
+    expect(screen.getByRole("textbox", { name: /firstName/i }).value).toBe(
+      "name"
+    );
+    expect(screen.getByRole("textbox", { name: /lastName/i }).value).toBe(
+      "surname"
+    );
+    expect(screen.getByRole("textbox", { name: /email/i }).value).toBe(
+      "test@mail.com"
+    );
+    expect(screen.getByPlaceholderText(/^password/i).value).toBe(
+      `${"a".repeat(MIN_CHAR - 1)}`
+    );
+    expect(screen.getByPlaceholderText(/repeat password/i).value).toBe(
+      `${"a".repeat(MIN_CHAR - 1)}`
+    );
+    expect(screen.getByRole("checkbox").checked).toBe(true);
+  });
+
+  it("should display max length error when password is long", async () => {
+    fireEvent.input(screen.getByRole("textbox", { name: /firstName/i }), {
```

Comment:
not sure why this is in this test but if you are relying on the order of the fields for testing practice this is also not great structuring.


imagine that I move the first name field below the password field. if this breaks a lot of tests and I have to go through each one of your unit tests to fix them, then something needs to be changed either about the structure of the code (not sure if it's showing one error at a time right now)

basically as a reader I am reading a test about max length of password and I see a lot of stuff about first and last name and I'm confused

Reply:
First, filling the input fields does not relay on any order, so I will try to move the input field we are focusing on and testing to the top of the unite. Second, I have to insert a valid value into those filed other that the field that I am testing to check that there is just one error. For error messages shown, we are showing just one error message at a time.